components/calendar/src/cal/persian.rs

```diff
@@ -84,32 +87,64 @@ impl CalendarArithmetic for Persian {
     }
 }
 
+impl DateFieldsResolver for Persian {
+    type YearInfo = i32;
+
+    #[inline]
+    fn year_info_from_era(&self, era: &str, era_year: i32) -> Result<Self::YearInfo, DateError> {
+        match era {
+            "ap" | "sh" | "hs" => Ok(era_year),
+            _ => Err(DateError::UnknownEra),
+        }
+    }
+
+    #[inline]
+    fn year_info_from_extended(&self, extended_year: i32) -> Self::YearInfo {
+        extended_year
+    }
+
+    #[inline]
+    fn reference_year_from_month_day(
+        &self,
+        month_code: types::MonthCode,
+        day: u8,
+    ) -> Result<Self::YearInfo, DateError> {
+        let (ordinal_month, _is_leap) = month_code
+            .parsed()
+            .ok_or(DateError::UnknownMonthCode(month_code))?;
+        // December 31, 1972 occurs on 10th month, 10th day, 1351 AP
```

Comment:
you have a lot of these claims but it's not clear that this is correct. is this claim tested? can you use const to actually compute this from the 1972-12-31?

Reply:
We can't use const for non-ISO stuff. These claims can be produced by swapping the calendars in the code linked elsewhere, perhaps we should link it here too.

The nearby-range validity check should ensure that these dates are valid. Code that generates this info would be nice.